driver/src/test/java/oracle/nosql/driver/OnPremiseTest.java

```diff
@@ -518,20 +540,82 @@ public void testNamespaces()
+
+            /* query child with explicit namespace */
+            query = "select * from " + nsChildName;
+            res = doQuery(handle, query);
+            assertEquals(numParent * numChild, res.size());
+
         } catch (RequestTimeoutException rte) {
-            /*
-            if (!(rte.getCause() instanceof SystemException)) {
+            if (checkKVVersion(20, 1, 1) ||
+                !(rte.getCause() instanceof SystemException)) {
                 throw rte;
             }
-            */
             /* ignore this exception for 19 for now; known bug */
         }
 
-        /* test ListTables with namespace */
-        ListTablesRequest listTables =
-            new ListTablesRequest().setNamespace("myns");
-        ListTablesResult lres = handle.listTables(listTables);
-        assertEquals(2, lres.getTables().length);
+        /* Verify per-request namespaces */
+        ((NoSQLHandleImpl)handle).setDefaultNamespace(null);
+
+        /* this should fail */
+        getReq = new GetRequest().setTableName(parentName)
+            .setKey(new MapValue().put("id", 1));
+        try {
+            getRes = handle.get(getReq);
+            fail("Get should have failed");
+        } catch (TableNotFoundException e) {
+            // expected
+        }
+
+        /* This should pass */
+        getReq = new GetRequest().setTableName(parentName)
+            .setNamespace("myns")
+            .setKey(new MapValue().put("id", 1));
+        getRes = handle.get(getReq);
+        assertNotNull(getRes.getValue());
+
+        /* This should pass */
+        ((NoSQLHandleImpl)handle).setDefaultNamespace("invalid");
+        getReq = new GetRequest().setTableName(parentName)
+            .setNamespace("myns")
+            .setKey(new MapValue().put("id", 1));
+        getRes = handle.get(getReq);
+        assertNotNull(getRes.getValue());
+
+        /* verify table name overrides request namespace */
+
+        /* this should fail */
+        getReq = new GetRequest().setTableName(parentName)
+            .setNamespace("invalid")
+            .setKey(new MapValue().put("id", 1));
+        try {
+            getRes = handle.get(getReq);
+            fail("Get should have failed");
+        } catch (TableNotFoundException e) {
+            // expected
+        }
+
+        /* This should pass */
+        getReq = new GetRequest().setTableName(nsParentName)
+            .setNamespace("invalid")
+            .setKey(new MapValue().put("id", 1));
+        getRes = handle.get(getReq);
+        assertNotNull(getRes.getValue());
+
+        /* drop table with namespace in request */
+        treq = new TableRequest()
+            .setStatement("drop table parent.child")
+            .setNamespace("myns");
+        tres = handle.tableRequest(treq);
+        tres.waitForCompletion(handle, 100000, 1000);
+
+        /* drop namespace - use cascade to remove tables */
+        doSysOp(handle, "drop namespace myns cascade");
```

Comment:
Maybe verify that the tables and the namespace is gone, with listTables() and listNamespaces()?

Reply:
Not sure that really matters for this test, as the @After logic will delete the namespace and tables as well. I added this just to bring it in line with the internal proxy namespace tests. Note it will throw an error if the drop fails.